src/clojure_boost/week3/schemata/compra_client.clj

```diff
@@ -0,0 +1,175 @@
+(ns clojure-boost.week3.schemata.compra_client
+  (:use clojure.pprint)
+  (:require [schema.core :as s]
+            [clojure-boost.week1.lista-compras :as lista]))
+
+; É um caminho de validação também, mas pode melhorar utilizando validação por :atributo  e valor
+(s/set-fn-validation! true)
+
+(def CompraSchema {:id              s/Num
+                   :data            s/Str
+                   :valor           s/Num
+                   :estabelecimento s/Str
+                   :categoria       s/Str
+                   :cartão          s/Num})
+
+; Ele explica o que é aquele schaema
+;(comment (pprint (s/explain CompraSchema)))
+
+(s/def compra :- CompraSchema {:id              01
+                               :data            "2022/01/01"
+                               :valor           100M
+                               :estabelecimento "Amazon"
+                               :categoria       "Casa"
+                               :cartão          1234123412341234})
+
+
+;---------------------------------------------------------------------------------------------------
+
+ ;Validacao por atributo
+(s/set-fn-validation! true)
+
+; valida categoria
+(def testes ["Alimentacao", "Automóvel", "Casa", "Educação", "Lazer", "Saúde"])
+(get testes 1)
+
+(defn valida-categoria [categoria]
+  (if (nil? (some #(= categoria %) testes)) false true)); perguntar para o Cácio!!
+
+(valida-categoria "Ifood")
+
+(defn dois-caracteres? [estabelecimento]
+  (>= (count estabelecimento) 2))
+
+;(dois-caracteres? "n")
+
+(defn cartao-valido-Int? [cartao]
+  (and (> cartao 0)
+       (< cartao 10000000000000000)))
+
+(defn dataFormat? [data]
+  (re-matches #"\d{4}-\d{2}-\d{2}" data))
+
+(def ValidaData (s/constrained s/Str dataFormat?))
+(def ValidaId (s/pred pos-int?))
```

Comment:
Dessa forma, toda compra, obrigatoriamente, precisa ter o ID preenchido. Se a estratégia for delegar essa tarefa para o banco de dados, talvez o schema precisasse permitir `nil`. Mas é decisão de design. ;)